src/System.Linq/src/System/Linq/Enumerable.Sum.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace System.Linq
+{
+    public static partial class Enumerable
+    {
+        public static int Sum(this IEnumerable<int> source)
+        {
+            if (source == null) throw Error.ArgumentNull("source");
+            int sum = 0;
+            checked
+            {
+                foreach (int v in source) sum += v;
+            }
+            return sum;
+        }
+
+        public static int? Sum(this IEnumerable<int?> source)
+        {
+            if (source == null) throw Error.ArgumentNull("source");
+            int sum = 0;
+            checked
+            {
+                foreach (int? v in source)
+                {
+                    if (v != null) sum += v.GetValueOrDefault();
```

Comment:
I didn't change that. Just extracted from `Enumerable.cs` to `Enumerable.Sum.cs`. But agree, we might want to remove the `if` statement as part of the change.